libqtile/utils.py

```diff
@@ -219,22 +230,19 @@ def class_proxy(*args, **kwargs):
         globals_[class_name] = class_proxy
 
 
-def send_notification(title, message, urgent=False, timeout=10000):
+def send_notification(title, message, urgent=False, timeout=10000, id=None):
     """Send a notification."""
-    try:
-        import gi
-        gi.require_version("Notify", "0.7")
-        from gi.repository import Notify
-        Notify.init("Qtile")
-        info = Notify.get_server_info()
-        if info[0]:
-            notifier = Notify.Notification.new(title, message)
-            notifier.set_timeout(timeout)
-            if urgent:
-                notifier.set_urgency(Notify.Urgency.CRITICAL)
-            notifier.show()
-    except Exception as exception:
-        logger.error(exception)
+    if _can_notify and Notify.get_server_info()[0]:
+        notifier = Notify.Notification.new(title, message)
+        notifier.set_timeout(timeout)
+        if urgent:
+            notifier.set_urgency(Notify.Urgency.CRITICAL)
+        notifier.set_timeout(timeout)
+        if id is None:
+            id = randint(10, 1000)
```

Comment:
I'm probably overthinking this (especially since we didn't have an id before^^), but maybe a uuid or a timestamp would be nicer here?

Reply:
This is per the spec (see https://developer.gnome.org/notification-spec/). The Replaces ID is a UINT32.

Reply:
You always have those links at the ready.^^ Maybe add it in a comment?

Reply:
Sure, and I've explained how this can be used :)

Reply:
That's great stuff, thanks. :)